include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl.h

```diff
@@ -1466,6 +1466,119 @@ struct __partial_merge_kernel
     }
 };
 
+//Searching for an intersection of virtual matrix (n1, n2) diagonal with the Merge Path to define sub-ranges
+//to serial merge. For example, a virtual matrix for [0,1,1,2,3] and [0,0,2,3] shown below:
+//     0   1  1  2   3
+//    ------------------
+//   |--->
+// 0 | 0 | 1  1  1   1
+//   |   |
+// 0 | 0 | 1  1  1   1
+//   |   ---------->
+// 2 | 0   0  0  0 | 1
+//   |             ---->
+// 3 | 0   0  0  0   0 |
+template<typename Rng1, typename Rng2, typename _Index, typename _Index1, typename _Index2>
+auto __find_start_point(Rng1& __rng1, Rng2& __rng2, _Index __i_elem, _Index1 __n_1, _Index2 __n_2)
+{
+    _Index1 __end1 = 0;
+    _Index2 __end2 = 0;
+    if (__i_elem < __n_1) //a condition to specify upper or down part of the merge matrix to be processed
+    {
+        auto __q = __i_elem; //digonal index
+        auto __n_diag = __q < __n_2 ? __q : __n_2; //digonal size
+
+        oneapi::dpl::counting_iterator<_Index> __diag_it(0);
+        auto __res = std::lower_bound(__diag_it, __diag_it + __n_diag, 1, [&__rng1, &__rng2, __q](auto __v1, auto __v2)
+            {
+                auto __val1 = __rng1[__q - __v1 - 1] < __rng2[__v1];
+                return __val1 < __v2;
+            });
+        __end1 = __q - *__res;
+        __end2 = *__res;
+    }
+    else
+    {
+        auto __q = __i_elem - __n_1; //diagonal index
+
+        auto __n_diag = __n_2 - __q;
+        __n_diag = __n_diag < __n_1 ? __n_diag : __n_1; //digonal size
+
+        counting_iterator<_Index> __diag_it(0);
+        auto __res = std::lower_bound(__diag_it, __diag_it + __n_diag, 1, [&__rng1, &__rng2, __n_1, __q](auto __v1, auto __v2)
+            {
+                auto __val1 = __rng1[__n_1 - __v1 - 1] < __rng2[__q + __v1];
+                return __val1 < __v2;
+            });
+
+        __end1 = __n_1 - *__res;
+        __end2 = __q + *__res;
+    }
+    return std::make_pair(__end1, __end2);
+}
+
+template<typename _Rng1, typename _Rng2, typename _Rng3, typename _Index1, typename _Index2, typename _Index3,
+         typename _Size, typename _Size1, typename _Size2, typename _Compare>
```

Comment:
what do u mean?

Reply:
```suggestion
template<typename _Rng1, typename _Rng2, typename _Rng3, typename _Index1, typename _Index2, typename _Index3,
         typename _Size1, typename _Size2, typename _Size3, typename _Compare>
```

For better consistency of naming.

Reply:
Yes, I can see your point, It would be bad to change the indices of the related template arguments to not match, or mislead that the chunk size as corresponding to the `_Rng3`.  

It may be more clear that it is intentional if you reorder the parameters to have `_Size __chunk` second to last.  However, I'm fine with however you decide to do this.

Reply:
"_Size __chunk second to last." also breakers a call semantic... 
Indeed,
```
__serial_merge(const _Rng1& __rng1, const _Rng2& __rng2, _Rng3& __rng3, _Index1 __start1, _Index2 __start2,
               _Index3 __start3, _Size __chunk, _Size1 __n1, _Size2 __n2, _Compare __comp)
```
Tells:
"Does serial merge the data from rng1(starting from start1) and rng2(starting from start2) and writting to rng3 (starting from start3) in '__chunk' steps , but doesn't exceed of total size of  sequences  - n1 and n2."

May be it is better if '__chunk' will be at the first palce:
"Does  '__chunk' steps of serial merging of  the data from rng1 and rng2....."